### PR TITLE
Baseline implementation of MermaidJS Output Formatter - qossmic/deptrac#1372

### PR DESCRIPTION
**Pull request by jan-simbera**
The basic implementation of the MermaidJS Output Formatter. With documentation and php config.

### CODE REVIEW

**Comment by patrickkusebauch** [comment]
Closes: https://github.com/qossmic/deptrac/issues/1372

For cross-referencing, can you add this to the PR description, please?

File: src/Contract/Config/Formatter/MermaidJsConfig.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace Qossmic\Deptrac\Contract\Config\Formatter;
+
+use Qossmic\Deptrac\Contract\Config\Layer;
+
+final class MermaidJsConfig implements FormatterConfigInterface
+{
+    private string $name = 'mermaidjs';
+
+    private string $direction = 'TD';
```
Comment (patrickkusebauch) [inline src/Contract/Config/Formatter/MermaidJsConfig.php:13]: In the `yaml` config, you allow to modify this parameter, but I do not see an equivalent way to do this in the `php` config. Can you make sure the feature set is equivalent between the two?

File: docs/formatters.md [hunk 2]
```diff
@@ -141,6 +141,12 @@ Will produce the following graph:
 #### Pointing to groups instead of nodes
 With `formatters.graphviz.pointToGroups` set to `true`, when you have a node inside a groups with the same name as the group itself, edges pointing to that node will point to the group instead. This might be useful for example if you want to provide a "public API" for a module defined by a group.
 
+## MermaidJS Formatter
+
+The MermaidJS formatter is a console formatter, which generates a mermaid.js compatible graph definition. It can be activated with `--formatter=mermaidjs`.
+With the -o option you can specify the output file.
+
```
Comment (patrickkusebauch) [inline docs/formatters.md:148]: Can you provide and example here similar to other formatters?

**Review by patrickkusebauch — CHANGES_REQUESTED** [review]

File: src/Contract/Config/Formatter/MermaidJsConfig.php (hunk 1 above)
Comment (jan-simbera) [reply src/Contract/Config/Formatter/MermaidJsConfig.php:13]: Should be now equivalent.

File: docs/formatters.md (hunk 2 above)
Comment (jan-simbera) [reply docs/formatters.md:148]: I have extended the documentation in the README.

**Comment by patrickkusebauch** [comment]
> Closes: https://github.com/qossmic/deptrac/issues/1372

Oh not to the title, to the description. Literally just copy and paste it as the first line in the description.

**Review by patrickkusebauch — APPROVED** [review]
:+1: **LGTM**

**Comment by gennadigennadigennadi** [comment]
@jan-simbera thank you very much!

**Pull request merged**